salt/state.py

```diff
@@ -149,6 +150,13 @@ def _gen_tag(low):
     return u'{0[state]}_|-{0[__id__]}_|-{0[name]}_|-{0[fun]}'.format(low)
 
 
+def _clean_tag(tag):
+    '''
+      urllib safe quote the tag value to avoid invalid chars in the filename
+    '''
+    return urllib.quote(tag.encode('utf8'), safe='')
```

Comment:
Yeah I'm actually planning on using this method:
https://github.com/saltstack/salt/pull/43056/files#diff-bdabb2b499cd3b4adf29d764d4eec419R168